src/tester.rs

```diff
@@ -225,22 +231,24 @@ impl<'a> LangTester<'a> {
 
     /// Enumerate all the test files we need to check, along with the number of files filtered out
     /// (e.g. if you have tests `a, b, c` and the user does something like `cargo test b`, 2 tests
-    /// (`a` and `c`) will be filtered out.
+    /// (`a` and `c`) will be filtered out. The `PathBuf`s returned are guaranteed to be fully
+    /// canonicalised.
     fn test_files(&self) -> (Vec<PathBuf>, usize) {
         let mut num_filtered = 0;
-        let paths = WalkDir::new(self.test_dir.unwrap())
+        let paths = WalkDir::new(self.inner.test_dir.as_ref().unwrap())
             .into_iter()
             .filter_map(|x| x.ok())
             .filter(|x| x.file_type().is_file())
+            .map(|x| canonicalize(x.into_path()).unwrap())
```

Comment:
Do we `unwrap` here because the paths are expected to have already been passed to `canonicalize` before (and hence, this is guaranteed not to fail)?

Reply:
This isn't guaranteed to fail, but I've never seen one of these calls fail, so I'm not really sure what to do / report when it does fail. In retrospect, `lang_tester` should probably use `Result` in return types more often, but I'm not sure it's worth changing the API now...

Reply:
"isn't guaranteed not to fail" even...

Reply:
Guarantee was a poor word choice. What I meant was on line 84 you have:
```rust          
inner.test_dir = Some(canonicalize(test_dir).expect(&format!(...
```

If this is always run first, then you're expecting `canonicalize` not to fail the second time around too?

If that's not the case, perhaps this `unwrap` should be replaced by `expect` in the same way.

Reply:
Agreed, it *probably* won't fail. In general, I'm not sure `expect` is better than `unwrap`: it makes the code uglier, often for cases that we'll probably never see. I suspect this is one of them.

Reply:
Agreed. On that basis do you think that the `expect` on line 84 should just be an `unwrap` too?

Reply:
Good question. I assumed that there's a small possibility that one might fail, but I'm just as happy for it to be an `unwrap`! Thoughts?

Reply:
I prefer `unwrap` tbh. I'm not sure the `expect` adds much apart from ugliness - In the rare case it does fail, `unwrap` will give a line number and it will be easy to then see that it went wrong in some way.

Reply:
Fixed in https://github.com/softdevteam/lang_tester/pull/58/commits/340709b36e1a0f85aa210934ffa14ee50feb5751.